Emby.Server.Implementations/Data/SqliteItemRepository.cs

```diff
@@ -2914,29 +2914,29 @@ public QueryResult<BaseItem> GetItems(InternalItemsQuery query)
         private string GetOrderByText(InternalItemsQuery query)
         {
             var orderBy = query.OrderBy;
-            if (string.IsNullOrEmpty(query.SearchTerm))
+            bool hasSimilar = query.SimilarTo != null;
+            bool hasSearch = !string.IsNullOrEmpty(query.SearchTerm);
+
+            if (hasSimilar || hasSearch)
             {
-                int oldLen = orderBy.Count;
-                if (oldLen == 0 && query.SimilarTo != null)
+                List<(string, SortOrder)> prepend = new List<(string, SortOrder)>(4);
+                if (hasSearch)
                 {
-                    var arr = new (string, SortOrder)[oldLen + 2];
-                    orderBy.CopyTo(arr, 0);
-                    arr[oldLen] = ("SimilarityScore", SortOrder.Descending);
-                    arr[oldLen + 1] = (ItemSortBy.Random, SortOrder.Ascending);
-                    query.OrderBy = arr;
+                    prepend.Add(("SearchScore", SortOrder.Descending));
+                    prepend.Add((ItemSortBy.SortName, SortOrder.Ascending));
                 }
-            }
-            else
-            {
-                query.OrderBy = new[]
+                if (hasSimilar)
                 {
-                    ("SearchScore", SortOrder.Descending),
-                    (ItemSortBy.SortName, SortOrder.Ascending)
-                };
-            }
-
+                    prepend.Add(("SimilarityScore", SortOrder.Descending));
+                    prepend.Add((ItemSortBy.Random, SortOrder.Ascending));
+                }
 
-            if (orderBy.Count == 0)
+                var arr = new (string, SortOrder)[prepend.Count + orderBy.Count];
+                prepend.CopyTo(arr, 0);
+                orderBy.CopyTo(arr, prepend.Count);
+                orderBy = query.OrderBy = arr;
```

Comment:
Using an intermediate array here seems more difficult to read to me. What about:

```
query.OrderBy = prepend.Concat(query.OrderBy).ToArray();
```

I would also just get rid of the intermediate local variable `orderBy` entirely, it doesn't appear to serve much purpose.

Reply:
(`Concat` is SLOW)

Reply:
> (`Concat` is SLOW)

I prefer readability for small arrays. I think this would only ever be 5 or 6 elements here, so I would think the performance difference would be unnoticeable, if any

Reply:
We have a long standing disagreement about how to balance readability with performance, and I don't think we should block this pull request for that reason.